cmd/internal/kube/injector/injector.go

```diff
@@ -0,0 +1,204 @@
+			continue
+		}
+
+		var deployment *appsv1.Deployment
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &deployment)
+		if err != nil {
+			return nil, err
+		}
+
+		if deployment.Namespace == "" {
+			set["default"] = struct{}{}
+		} else {
+			set[deployment.Namespace] = struct{}{}
+		}
+	}
+
+	return sets.StringKeySet(set).List(), nil
+}
+
+func (i *injectorImpl) Inject(sidecar v1.Container) ([]*unstructured.Unstructured, error) {
+	onMap := func(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+		out := obj.DeepCopy()
+		gvk := out.GetObjectKind().GroupVersionKind()
+
+		if !isInjectable(gvk) {
+			return out, nil
+		}
+
+		var deployment *appsv1.Deployment
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(out.Object, &deployment)
+		if err != nil {
+			return nil, err
+		}
+
+		injectedDeployment := deployment.DeepCopy()
+		if err != nil {
+			return nil, err
+		}
+
+		containers := injectedDeployment.Spec.Template.Spec.Containers
+		injectedDeployment.Spec.Template.Spec.Containers = append(containers, sidecar)
+
+		if err != nil {
+			return nil, err
+		}
+
+		out.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(injectedDeployment)
+		if err != nil {
+			return nil, err
+		}
+
+		return out, nil
+	}
+
+	return mapUnstructured(i.objects, onMap)
+}
+
+func isInjectable(kind schema.GroupVersionKind) bool {
+	return kind.Group == "apps" && kind.Version == "v1" && kind.Kind == "Deployment"
+}
+
+// fromRawObject converts raw bytes into an unstructure.Unstrucutred object.
+// unstructured.Unstructured is used to represent a Kubernetes object that is not known ahead of time.
+func fromRawObject(raw []byte) (*unstructured.Unstructured, error) {
+	jConfigMap, err := kyamlutil.ToJSON(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	object, err := runtime.Decode(unstructured.UnstructuredJSONScheme, jConfigMap)
+	if err != nil {
+		return nil, err
+	}
+
+	unstruct, ok := object.(*unstructured.Unstructured)
+	if !ok {
+		return nil, errors.New("unstructured conversion failed")
+	}
+
+	return unstruct, nil
+}
+
+func getFile(filePath string) ([]byte, error) {
+	fileDir, fileName := filepath.Split(filePath)
+
+	absOutputDir, err := filepath.Abs(fileDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check for directory existence
+	if _, staterr := os.Stat(absOutputDir); os.IsNotExist(staterr) {
+		return nil, errors.Wrapf(staterr, "directory %s does not exist", absOutputDir)
+	}
+
+	absPath := filepath.Join(absOutputDir, fileName)
+
+	// Check for existence of file
+	if _, staterr := os.Stat(absPath); os.IsNotExist(staterr) {
+		return nil, errors.Wrapf(staterr, "file %s does not exist", absPath)
+	}
+
+	return os.ReadFile(absPath)
+}
+
+// mapUnstructured applies the given transformer function to each unstructured Kubernetes item in the given list.
+// If the transformer function returns an error, the error is returned immediately.
+func mapUnstructured(
```

Comment:
I like this utility function, but I think it could be replaces by akitasoftware/go-utils slices.MapWithErr?

https://github.com/akitasoftware/go-utils/blob/6e85caefbf2ff699b1923eac1807e14cbdc894e1/slices/map.go#L13

Reply:
I've updated the injector to use our existing libraries in [1b26251](https://github.com/akitasoftware/akita-cli/pull/207/commits/1b262511defc26e66c122ce152d99f196bf24e76)